oss_src/logger/logger.hpp

```diff
@@ -558,6 +572,8 @@ class file_logger{
   bool log_to_stderr;
   int log_level;
 
+  size_t reference_pid = 0;
+
```

Comment:
This should maybe start as size_t(-1), since 0 can be a valid (though unlikely) PID.